packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -208,15 +219,28 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     }
   }, [isOpen]);
 
-  // Initialize filtered options to the latest value options
+  // Update filtered options to the latest value options according to input value
   const [filteredOptions, setFilteredOptions] = React.useState(options);
   React.useEffect(() => {
-    setFilteredOptions(options);
+    if (inputValue) {
```

Comment:
will it always have the newest `inputValue` ?

Reply:
It should

Reply:
how can you be sure?

Reply:
So `useEffect` will trigger when `options` changes and then takes latest values of everything including `inputValue`